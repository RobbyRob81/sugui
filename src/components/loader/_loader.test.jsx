import * as React from 'react';
import { customRender } from '../../../test/test-utils';
import Loader from '.';

it('should render without throwing an error', () => {
  const { container } = customRender(<Loader isLoading={false} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('should be visible when loading prop is set to true', () => {
  const { getByTestId } = customRender(<Loader isLoading />);
  expect(getByTestId('loader')).toBeVisible();
});
