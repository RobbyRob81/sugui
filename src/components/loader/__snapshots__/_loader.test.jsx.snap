// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<section
  class="sc-bdVaJa NiJku"
  data-testid="loader"
>
  <div
    class="sc-bwzfXH lbGtxA"
  >
    <div
      class="sc-htpNat davlbj"
    />
    <div
      class="sc-EHOje dCSvZy"
    >
      <div
        class="sc-bZQynM fTnrNt"
      />
    </div>
    <div
      class="sc-bxivhb dZAzLp"
    >
      <div
        class="sc-ifAKCX fVKAHz"
      >
        Loading...
      </div>
    </div>
  </div>
  <button
    class="sc-gzVnrw hsCKaI"
  >
    <svg
      color="white"
      height="31"
      viewBox="0 0 32 31"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h32v32H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M26.667 5.333L5.333 26.667M5.333 5.333l21.334 21.334"
          stroke="white"
          stroke-linecap="round"
        />
      </g>
    </svg>
  </button>
</section>
`;
